src/sagemaker/serializers.py

```diff
@@ -43,6 +44,59 @@ def CONTENT_TYPE(self):
         """The MIME type of the data sent to the inference endpoint."""
 
 
+class NumpySerializer(BaseSerializer):
+    """Serialize data to a buffer using the .npy format."""
+
+    CONTENT_TYPE = "application/x-npy"
+
+    def __init__(self, dtype=None):
+        """Initialize the dtype.
+
+        Args:
+            dtype (str): The dtype of the data.
+        """
+        self.dtype = dtype
+
+    def serialize(self, data):
+        """Serialize data to a buffer using the .npy format.
+
+        Args:
+            data (object): Data to be serialized. Can be a NumPy array, list,
+                file, or buffer.
+
+        Returns:
+            io.BytesIO: A buffer containing data serialzied in the .npy format.
+        """
+        if isinstance(data, np.ndarray):
+            if not data.size > 0:
```

Comment:
this might be clearer as `if data.size == 0`? (it can't ever be negative, right? 😂)